### ISSUE TITLE
[css-break] Clarify how `box-shadow` should appear when a box is sliced

### ISSUE THREAD

**Issue by jfkthame** [w3c/csswg-drafts#10171]
The description at https://drafts.csswg.org/css-break/#valdef-box-decoration-break-slice does not seem to explicitly address the slicing of box shadows. There's an illustration there of how `box-decoration-break: slice` treats a _border_, which is straightforward, but it doesn't clarify how a _shadow_ should be treated.

See https://wpt.fyi/results/css/css-backgrounds/box-shadow/slice-block-fragmentation-001.html, which shows a test that currently passes in webkit and blink, but fails in gecko. IMO, gecko's result is what should happen, and the test expectation is incorrect, but I don't think the spec adequately addresses this.

When a box (with its decorations) is "sliced" into fragments, and its box-shadow is also "sliced", does the corresponding slicing of the shadow take account of its offset from the box? That's what gecko does, and to me it makes sense visually. Or does the shadow, although painted at an offset, get sliced at the same places as the box? That seems to be what the current test expects, but looks illogical.

With a more extreme example https://codepen.io/jfkthame/pen/yLrPrvr, where there's a simple yellow box that has a gray shadow, and then a cyan version of the box (with the same shadow) that is fragmented across columns, Firefox's rendering still looks reasonable, while both Chrome and Safari do things with the shadow of the sliced cyan box that just don't make any sense.

I'd like to adjust the WPT reference case to reflect the behavior I think is correct, but I also think an illustration in the spec to confirm this expectation would be useful.

**Comment by Loirooriol** [comment]
But in this case I prefer Blink:

```html
<!DOCTYPE html>
<style>
div {
  width: 0;
  line-height: 3;
}
span {
  box-shadow: 5px 5px 0 10px silver;
  height: 300px;
}
</style>
<div>aaaaaaaa<span>bbbbb ccccc</span>ddddd</div>
```

| Gecko | Blink | WebKit | Servo |
| - | - | - | - |
| ![](https://github.com/w3c/csswg-drafts/assets/7477678/26bb0679-e3ab-4c89-a09d-51fc1e59b9f9) | ![](https://github.com/w3c/csswg-drafts/assets/7477678/10242f8e-19eb-4909-8694-f35171330726) | ![imatge](https://github.com/w3c/csswg-drafts/assets/7477678/98be94f1-bd24-443c-893f-6615c0c42969) | ![](https://github.com/w3c/csswg-drafts/assets/7477678/ade35036-f9fd-4e54-95a9-3242a4f51d8c) |

**Comment by kizu** [comment]
@Loirooriol This feels like a case for `box-decoration-break: clone`: with it, in Gecko it is possible to do either:

https://codepen.io/kizu/pen/MWRQezr

<img width="153" alt="A screenshot of a CodePen with two wrapping inline elements that have a box-shadow on both fragments, in one case the box-shadow stops before and after wrapping, as if the element was sliced in two, in another case it is as if both fragments were independent." src="https://github.com/w3c/csswg-drafts/assets/177485/e76003a1-8c08-43cf-a77c-ab9b477d47c4">

(note that `box-decoration-break` also seems to work correctly in Gecko with the original example in this issue, and the rendering of that example in Gecko, indeed, makes much more sense.)

**Comment by Loirooriol** [comment]
@kizu No, I'm talking about `slice`. Gecko keeps the same shadow size as for `clone`, so it produces this weird spacing.

**Comment by kizu** [comment]
Ah, sorry, did not notice this difference. For this specific example, Blink's rendering looks better, indeed. It would be interesting to look at some other use cases with different shadows, but in general stopping the shadow earlier seems better for `slice`, yes.

**Comment by Loirooriol** [comment]
The spec says

> The effect is as though the element were rendered with no breaks present, and then sliced by the breaks afterward

This seems closer to Blink for my previous testcase, but here Blink produces a bigger shadow when slicing:

```html
<!DOCTYPE html>
<style>span { box-shadow: 200px 0px 0 1px silver }</style>
<div><span>aaaaaaaa bbbbbbbb</span></div>
<div style="width: 0"><span>aaaaaaaa bbbbbbbb</span></div>
```

![imatge](https://github.com/w3c/csswg-drafts/assets/7477678/d12021c9-9a49-479f-8037-28633a01ac3a)

**Comment by jfkthame** [comment]
I think this example has to be regarded as simply a Blink rendering bug, not a reasoned interpretation of the spec.

**Comment by benface** [comment]
I could be wrong, but based on browser inconsistencies, I believe the effect of `box-decoration-break` on `outline` is also not very well defined. See [this webkit "bug"](https://bugs.webkit.org/show_bug.cgi?id=258697).

**Comment by fantasai** [comment]
The core spec definition is "The effect is as though the element were rendered with no breaks present, and then sliced by the breaks afterward." So there should be no shadow beyond the breakpoint. And shadows don't take up space, so there should be no space left for them either.
